### ISSUE TITLE
bayes_builder improve

### ISSUE THREAD

**Issue by Chenliang-Zhou** [AISpace2/AISpace2#96]
- bayes builder improve:
	- UI
		- Select: show green "Updated." after submitting
		- Order: Creat -> Select -> Set CPT -> Delete
		- when change b/w Select and Set CPT, remain the selected item
	- add
		- not allow same value in the domain
		- when create edge: do not reset children's CPT, just copy the first line
		- combine Variable and Edge to be Create
	- delete edge
		- do not reset: set it to be the first one
	- set probability
		- (try) can input letters.....
		- precision
		- change the color enter (not until submit)
		- cannot show CPT after change name?
		- change self domain:
			- same number of values: unchanged (already done...)
			- more values: add 0
			- less values: normalization
		- change parent's domain:
			- same number of values: unchanged
			- more values: set only this line to default (uniform)
			- less values: just delete that line, others unchanged

**Comment by Chenliang-Zhou** [comment]
- When just open the builder, make default mode "Create" instead of "Select"

- After adding a node and then change to "Edge," clear the "Variable created." message

- not allow same value in the domain: please prompt a user with an error message saying "Duplicate value in domain is not allowed." (Like what you did for "Edge already exists.") instead of just ignoring the duplicate value.

- BUG in CPT: Cannot input 0.something (cannot input dot after 0)

- After deleting a parent, the CPT looks like this:
<img width="782" alt="屏幕快照 2019-07-22 10 10 28" src="https://user-images.githubusercontent.com/39361141/61650629-fba54780-ac68-11e9-9ab7-54d743b7d7cd.png">

but it should be just set the corresponding lines to the first one.

- Still can input letters and the error message does not match:
<img width="593" alt="屏幕快照 2019-07-22 10 12 29" src="https://user-images.githubusercontent.com/39361141/61650774-458e2d80-ac69-11e9-9c66-629c15002325.png">

- Delete: can you make a confirm message? (e.g. after the user chooses an object, show below the canvas: "Delete this node?" (or "Delete this edge?" or "Delete this node and all incident edges?") and two buttons "Yes" and "No")

- Accept empty input box as 0:
<img width="526" alt="image" src="https://user-images.githubusercontent.com/39361141/61651544-edf0c180-ac6a-11e9-99ac-d848b31107b2.png">


- In CPT can you make the background color of the input boxes (yellow or red) change dynamically as the user inputs? E.g.:
<img width="521" alt="image" src="https://user-images.githubusercontent.com/39361141/61651597-09f46300-ac6b-11e9-939a-effff5184e50.png">

<img width="527" alt="image" src="https://user-images.githubusercontent.com/39361141/61651626-1973ac00-ac6b-11e9-923e-6eb9e8b8f42c.png">

**Comment by Chenliang-Zhou** [comment]
- This is what it shows when I opened it: still "Select" is the default...:
<img width="854" alt="image" src="https://user-images.githubusercontent.com/39361141/61765814-7bc7cc00-ad93-11e9-98be-064745a5c059.png">

- Should be able to use other languages as values in a domain:
<img width="844" alt="image" src="https://user-images.githubusercontent.com/39361141/61765472-0b6c7b00-ad92-11e9-897d-af3561dcdcc3.png">

- Add a `Uniform` button besides each line of CPT that once clicked, set that line to be the uniform distribution.

**Comment by dsou** [comment]
> * This is what it shows when I opened it: still "Select" is the default...:
> 
> <img alt="image" width="854" src="https://user-images.githubusercontent.com/39361141/61765814-7bc7cc00-ad93-11e9-98be-064745a5c059.png">
> 

Please also replace BayesBuilderToolbar.vue

> * Still not able to input a dot (.) into an empty input box. (This sometimes happens and sometimes I can...)

Maybe you input a dot that is not "." , but  "。"?  Works fine at my end.

**Comment by Chenliang-Zhou** [comment]
- One-line CPT does not have `Uniform` button, but it should also have (to make the view consistent):
<img width="409" alt="image" src="https://user-images.githubusercontent.com/39361141/61847319-16411180-ae5f-11e9-8da2-b31ac8b01d29.png">

- `Uniform` button has height larger than the CPT line height, then when we have many lines, it looks like this:....
<img width="784" alt="image" src="https://user-images.githubusercontent.com/39361141/61847401-5c967080-ae5f-11e9-82b5-d8e7c418e74a.png">

**Comment by dsou** [comment]
> * One-line CPT does not have `Uniform` button, but it should also have (to make the view consistent):
> 
> <img alt="image" width="409" src="https://user-images.githubusercontent.com/39361141/61847319-16411180-ae5f-11e9-8da2-b31ac8b01d29.png">
> 

All Uniform button should be enough one-line CPT.

> * `Uniform` button has height larger than the CPT line height, then when we have many lines, it looks like this:....
> 
> <img alt="image" width="784" src="https://user-images.githubusercontent.com/39361141/61847401-5c967080-ae5f-11e9-82b5-d8e7c418e74a.png">

Fixed.

**Comment by Chenliang-Zhou** [comment]
- Combine style info to `style.css`, and comment if it is only for `BayesBuilder.vue`.

- We do still need the `Uniform` button for one-line CPT since this would make the visualization more consistent and decrease the borden recognizing for users.

- Combine two submodes in Create into one, like this:

**To create variable**: Set the name and the domain of the variable below, 
and then double click at a position on the canvas where you want the new node to be created.
Name: [input box]      Domain: [input box]
[success/warning message here for variable creation if any]

**To create edge**: Select a start node, then select an end node.
[success/warning message here for variable creation if any]
